src/server/render-widget.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
```

Comment:
Also, for me it's a big no to duplicate this file as it's so large / complex. I'll try and think of a way around it.